modules/View.js

```diff
@@ -696,6 +705,55 @@
     //************** Private methods **************//
 
     /**
+     * Attaches views using this.viewMap. The API for viewMap looks like:
+     * viewMap: {
+     *   foo: fooView,  // foo is injectionSite, fooView is the view
+         bar: 'barView',  // bar is injectionSite, 'barView' is a field on the view (view.barView)
+         baz: function() {  // baz is injectionSite
+           return this.bazView;  // the context 'this' is the view
+         },
+         taz: {  // if you want to pass in options, use a config object with 'view' and 'options'
+           view: (same as the three above: direct reference, string of view field, or function that return view),
+           options: {} // optional options
+         } 
+     * }
+     * To create dynamic show/hide logic, perform the logic in a function that returns the correct view, or you can
+     * call this.set('hide:foo', true) or this.set('hide:foo', false)
+     * @private
+     * @method __attachViewsFromViewMap
+     */
+    __attachViewsFromViewMap: function() {
+      var viewMap = _.result(this, 'viewMap');
+      _.each(viewMap, function(config, injectionSiteName) {
+        if (!this.get('hide:' + injectionSiteName)) {
+          var options = {};
+          var trackedView;
+          if (_.isFunction(config)) {
+            config = config.call(this);
+          }
+          if (config instanceof Backbone.View) {
+            trackedView = config;
+          } else if (_.isObject(config)) {
+            options = config.options;
+            config = config.view;
+          }
+          if (!trackedView) {
+            if (_.isString(config)) {
```

Comment:
maybe order this chain of ifs in most likely to least?  To prevent checking unlikely things.  Unless it already is, I didn't really think about it.

Reply:
done